### PR TITLE
Avoid unnecessarily collecting a _fetchId in certain cases

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: sql/src/main/java/io/crate/planner/operators/Collect.java [hunk 1]
```diff
@@ -135,6 +137,37 @@
...
+    /**
+     * Return columns which are not used.
+     *
+     * Examples:
+     *
+     * <pre>
+     * toCollect: [f(x)]        used: x
+     * unused:    []
+     *
+     * toCollect: [x, f(x)]     used: [f(x)]
```
Comment (mxm) [inline sql/src/main/java/io/crate/planner/operators/Collect.java]: Should used be `x` here?

Comment (mfussenegger) [reply sql/src/main/java/io/crate/planner/operators/Collect.java]: That was deliberate to show that `used` my also contain scalars. (Ex. in the case of `ORDER BY x + 1`)

Comment (mxm) [reply sql/src/main/java/io/crate/planner/operators/Collect.java]: That makes sense. Still, wouldn't the algorithm output `[x, f(x)]`?

Comment (mfussenegger) [reply sql/src/main/java/io/crate/planner/operators/Collect.java]: no, used is empty in this case - I've added a test for this case

Comment (mxm) [reply sql/src/main/java/io/crate/planner/operators/Collect.java]: >no, used is empty in this case

Why is the example then still saying `f(x)`?

Comment (mfussenegger) [reply sql/src/main/java/io/crate/planner/operators/Collect.java]: Sorry, I meant that, `unused` is empty

**Pull request merged**